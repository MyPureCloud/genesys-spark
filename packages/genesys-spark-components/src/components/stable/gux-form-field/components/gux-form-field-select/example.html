<h1>gux-form-field-select</h1>

<p>slot a select element into a gux-form-field-select element</p>

<h2>Examples</h2>
<form onchange="notify(event)" oninput="notify(event)">
  <fieldset>
    <legend>Label Position</legend>

    <gux-form-field-select>
      <select slot="input" name="lp-1">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Default</label>
    </gux-form-field-select>

    <gux-form-field-select>
      <select slot="input" name="lp-2">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Default long label</label>
    </gux-form-field-select>

    <gux-form-field-select label-position="above">
      <select slot="input" name="lp-3">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Above</label>
    </gux-form-field-select>

    <gux-form-field-select label-position="beside">
      <select slot="input" name="lp-4">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Beside</label>
    </gux-form-field-select>

    <gux-form-field-select label-position="screenreader">
      <select slot="input" name="lp-5">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Screenreader</label>
    </gux-form-field-select>
  </fieldset>

  <fieldset>
    <legend>Input attributes</legend>

    <gux-form-field-select>
      <select slot="input" name="a-1" disabled>
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Disabled</label>
    </gux-form-field-select>

    <gux-form-field-select>
      <select slot="input" name="a-2" required>
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Required</label>
    </gux-form-field-select>

    <gux-form-field-select>
      <select slot="input" name="a-3">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3" selected>Option 3</option>
      </select>
      <label slot="label">Preset</label>
    </gux-form-field-select>
  </fieldset>

  <fieldset>
    <legend>Error</legend>

    <gux-form-field-select>
      <select slot="input" name="e-1">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Default</label>
      <span slot="error">This is an error message</span>
    </gux-form-field-select>
  </fieldset>

  <fieldset>
    <legend>Help</legend>

    <gux-form-field-select>
      <select slot="input" name="e-1">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Default</label>
      <span slot="help">This is a help message</span>
    </gux-form-field-select>
  </fieldset>
  <fieldset>
    <legend>Label Info</legend>

    <gux-form-field-select>
      <select slot="input" name="e-1">
        <option value="option1">Option 1</option>
        <option value="option2">Option 2</option>
        <option value="option3">Option 3</option>
      </select>
      <label slot="label">Default</label>
      <gux-label-info-beta slot="label-info">
        <span slot="content">This is some tooltip text</span>
      </gux-label-info-beta>
    </gux-form-field-select>
  </fieldset>
</form>

<style>
  gux-form-field-select {
    margin: 10px;
  }
</style>
