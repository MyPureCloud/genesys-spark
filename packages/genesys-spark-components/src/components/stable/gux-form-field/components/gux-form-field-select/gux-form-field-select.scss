@use '../../gux-form-field.scss';
@use '../../functional-components/gux-form-field-container/gux-form-field-container.scss';
@use '../../functional-components/gux-form-field-error/gux-form-field-error.scss';
@use '../../functional-components/gux-form-field-label/gux-form-field-label.scss';
@use '../../functional-components/gux-form-field-help/gux-form-field-help.scss';

@include gux-form-field-container.Style;
@include gux-form-field-error.Style;
@include gux-form-field-label.Style;
@include gux-form-field-help.Style;

:host {
  display: block;
}

::slotted(label) {
  @include gux-form-field.gse-ui-formControl-label-bold;
}

::slotted(select) {
  flex: 1 1 auto;
  align-self: auto;
  order: 0;
  width: 100%;
  height: var(--gse-ui-formControl-input-textfield-height);
  padding-left: var(--gse-ui-formControl-input-gap);
  color: var(--gse-ui-formControl-input-populatedColor);
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--gse-ui-formControl-input-backgroundColor);
  border: none;
  border-radius: var(--gse-ui-formControl-input-borderRadius);
  outline: none;

  &::placeholder {
    color: var(--gse-ui-formControl-input-placeholderColor);
    opacity: 1; // missing token COMUI-2187
  }
}

.gux-input-and-error-container {
  flex-grow: 1;

  .gux-input {
    .gux-input-container {
      position: relative;
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      gap: var(--gse-ui-formControl-input-gap);
      place-content: stretch center;
      align-items: center;
      width: 100%;
      font-family: var(--gse-ui-formControl-input-contentText-fontFamily);
      font-size: var(--gse-ui-formControl-input-contentText-fontSize);
      font-weight: var(--gse-ui-formControl-input-contentText-fontWeight);
      line-height: var(--gse-ui-formControl-input-contentText-lineHeight);
      color: var(--gse-ui-formControl-input-populatedColor);
      background-color: var(--gse-ui-formControl-input-backgroundColor);
      background-image: none;
      border: var(--gse-ui-formControl-input-default-border-width)
        var(--gse-ui-formControl-input-default-border-style)
        var(--gse-ui-formControl-input-default-border-color);
      border-radius: var(--gse-ui-formControl-input-borderRadius);

      &.gux-disabled {
        pointer-events: none;
        opacity: var(--gse-ui-formControl-input-disabled-opacity);
      }

      &:hover {
        gux-icon {
          color: var(--gse-ui-formControl-input-inputIcon-iconEndColor);
        }
      }

      gux-icon {
        position: absolute;
        top: 0;
        right: 0;
        width: var(--gse-ui-formControl-input-inputIcon-size);
        height: var(--gse-ui-formControl-input-inputIcon-size);
        margin: var(--gse-ui-formControl-input-padding);
        pointer-events: none;
      }

      &:focus-within {
        border: var(--gse-ui-formControl-input-active-border-width)
          var(--gse-ui-formControl-input-active-border-style)
          var(--gse-ui-formControl-input-active-border-color);
        border-radius: var(--gse-ui-formControl-focusRing-borderRadius);
        outline: var(--gse-ui-formControl-input-focus-border-width)
          var(--gse-ui-formControl-input-focus-border-style)
          var(--gse-ui-formControl-input-focus-border-color);
        outline-offset: var(--gse-semantic-focusRing-offset);
      }
    }

    &.gux-input-error {
      .gux-input-container {
        border-color: var(--gse-ui-formControl-input-error-border-color);
      }
    }
  }
}
