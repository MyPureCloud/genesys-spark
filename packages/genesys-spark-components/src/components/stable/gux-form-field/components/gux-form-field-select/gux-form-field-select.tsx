import { Component, Element, h, JSX, Prop, State } from '@stencil/core';

import { calculateInputDisabledState } from '@utils/dom/calculate-input-disabled-state';
import { onInputDisabledStateChange } from '@utils/dom/on-input-disabled-state-change';
import { OnMutation } from '@utils/decorator/on-mutation';
import { onRequiredChange } from '@utils/dom/on-attribute-change';
import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
import { hasSlot } from '@utils/dom/has-slot';

import {
  GuxFormFieldHelp,
  GuxFormFieldError,
  GuxFormFieldLabel,
  GuxFormFieldContainer
} from '../../functional-components/functional-components';

import {
  GuxFormFieldLabelPosition,
  GuxFormFieldIndicatorMark
} from '../../gux-form-field.types';
import {
  getComputedLabelPosition,
  validateFormIds
} from '../../gux-form-field.service';
import { trackComponent } from '@utils/tracking/usage';

/**
 * @slot input - Required slot for input tag
 * @slot label - Required slot for label tag
 * @slot error - Optional slot for error message
 * @slot help - Optional slot for help message
 */
@Component({
  styleUrl: 'gux-form-field-select.scss',
  tag: 'gux-form-field-select',
  shadow: true
})
export class GuxFormFieldSelect {
  private input: HTMLSelectElement;
  private label: HTMLLabelElement;
  private disabledObserver: MutationObserver;
  private requiredObserver: MutationObserver;

  @Element()
  private root: HTMLElement;

  @Prop()
  labelPosition: GuxFormFieldLabelPosition;

  /**
   * Field indicator mark which can show *, (optional) or blank
   * Defaults to required. When set to required, the component will display * for required fields and blank for optional
   * When set to optional, the component will display (optional) for optional and blank for required.
   */
  @Prop()
  indicatorMark: GuxFormFieldIndicatorMark = 'required';

  @State()
  private computedLabelPosition: GuxFormFieldLabelPosition = 'above';

  @State()
  private disabled: boolean;

  @State()
  private required: boolean;

  @State()
  private hasError: boolean = false;

  @State()
  private hasHelp: boolean = false;

  @OnMutation({ childList: true, subtree: true })
  onMutation(): void {
    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');
  }

  componentWillLoad(): void {
    this.setInput();
    this.setLabel();

    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');

    trackComponent(this.root, { variant: this.variant });
  }

  disconnectedCallback(): void {
    if (this.disabledObserver) {
      this.disabledObserver.disconnect();
    }
    if (this.requiredObserver) {
      this.requiredObserver.disconnect();
    }
  }

  render(): JSX.Element {
    return (
      <GuxFormFieldContainer labelPosition={this.computedLabelPosition}>
        <GuxFormFieldLabel
          position={this.computedLabelPosition}
          required={this.required}
          indicatorMark={this.indicatorMark}
        >
          <slot name="label" onSlotchange={() => this.setLabel()} />
        </GuxFormFieldLabel>
        <div class="gux-input-and-error-container">
          <div
            class={{
              'gux-input': true,
              'gux-input-error': this.hasError
            }}
          >
            <div
              class={{
                'gux-input-container': true,
                'gux-disabled': this.disabled
              }}
            >
              <slot name="input" />
              <gux-icon
                icon-name="custom/chevron-down-small-regular"
                decorative
              ></gux-icon>
            </div>
          </div>
          <GuxFormFieldError show={this.hasError}>
            <slot name="error" />
          </GuxFormFieldError>
          <GuxFormFieldHelp show={!this.hasError && this.hasHelp}>
            <slot name="help" />
          </GuxFormFieldHelp>
        </div>
      </GuxFormFieldContainer>
    ) as JSX.Element;
  }

  private get variant(): string {
    const labelPositionVariant = this.labelPosition
      ? this.labelPosition.toLowerCase()
      : 'none';

    const type = this.input.getAttribute('type');

    return `${type}-${labelPositionVariant}`;
  }

  private setInput(): void {
    this.input = this.root.querySelector('select[slot="input"]');

    preventBrowserValidationStyling(this.input);

    this.disabled = calculateInputDisabledState(this.input);
    this.required = this.input.required;

    this.disabledObserver = onInputDisabledStateChange(
      this.input,
      (disabled: boolean) => {
        this.disabled = disabled;
      }
    );
    this.requiredObserver = onRequiredChange(
      this.input,
      (required: boolean) => {
        this.required = required;
      }
    );

    validateFormIds(this.root, this.input);
  }

  private setLabel(): void {
    this.label = this.root.querySelector('label[slot="label"]');

    this.computedLabelPosition = getComputedLabelPosition(
      this.label,
      this.labelPosition
    );
  }
}
