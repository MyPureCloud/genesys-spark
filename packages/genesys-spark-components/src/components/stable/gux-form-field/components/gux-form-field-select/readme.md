# gux-form-field-select



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description | Type                                    | Default     |
| --------------- | ---------------- | ----------- | --------------------------------------- | ----------- |
| `labelPosition` | `label-position` |             | `"above" \| "beside" \| "screenreader"` | `undefined` |


## Slots

| Slot      | Description                     |
| --------- | ------------------------------- |
| `"error"` | Optional slot for error message |
| `"help"`  | Optional slot for help message  |
| `"input"` | Required slot for input tag     |
| `"label"` | Required slot for label tag     |


## Dependencies

### Depends on

- [gux-icon](../../../gux-icon)
- [gux-form-field-label-indicator](../../helper-components/gux-form-field-optional-indicator)

### Graph
```mermaid
graph TD;
  gux-form-field-select --> gux-icon
  gux-form-field-select --> gux-form-field-label-indicator
  style gux-form-field-select fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
