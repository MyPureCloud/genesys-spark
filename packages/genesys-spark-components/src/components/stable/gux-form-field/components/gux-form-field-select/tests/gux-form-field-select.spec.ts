import { newSpecPage } from '@test/specTestUtils';

import { GuxFormFieldSelect } from '../gux-form-field-select';

const components = [GuxFormFieldSelect];
const language = 'en';

describe('gux-form-field-select', () => {
  it('should build', async () => {
    const html = `
      <gux-form-field-select>
        <select slot="input" name="lp-1">
          <option value="option1">Option 1</option>
          <option value="option2">Option 2</option>
          <option value="option3">Option 3</option>
        </select>
        <label slot="label">Default</label>
      </gux-form-field-select>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxFormFieldSelect);
  });

  describe('#render', () => {
    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-select ${componentAttribute}>
              <select slot="input" name="lp-1">
                <option value="option1">Option 1</option>
                <option value="option2">Option 2</option>
                <option value="option3">Option 3</option>
              </select>
              <label slot="label">Default</label>
            </gux-form-field-select>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('input attributes', () => {
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-select>
              <select slot="input" name="lp-1" ${inputAttribute}>
                <option value="option1">Option 1</option>
                <option value="option2">Option 2</option>
                <option value="option3">Option 3</option>
              </select>
              <label slot="label">Default</label>
            </gux-form-field-select>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });
    describe('help', () => {
      it('should render component as expected', async () => {
        const html = `
        <gux-form-field-select>
        <select slot="input" name="e-1">
          <option value="option1">Option 1</option>
          <option value="option2">Option 2</option>
          <option value="option3">Option 3</option>
        </select>
        <label slot="label">Default</label>
        <span slot="help">This is a help message</span>
      </gux-form-field-select>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    describe('label-info', () => {
      it('should render component as expected', async () => {
        const html = `
        <gux-form-field-select>
        <select slot="input" name="e-1">
          <option value="option1">Option 1</option>
          <option value="option2">Option 2</option>
          <option value="option3">Option 3</option>
        </select>
        <label slot="label">Default</label>
        <gux-label-info-beta slot="label-info">
          <span slot="content">This is some tooltip text</span>
        </gux-label-info-beta>
      </gux-form-field-select>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
