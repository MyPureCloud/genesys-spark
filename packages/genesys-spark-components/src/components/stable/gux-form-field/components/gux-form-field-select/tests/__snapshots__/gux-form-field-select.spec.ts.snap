// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-select #render help should render component as expected 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help gux-show">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select aria-describedby="gux-form-field-help-i" id="gux-form-field-input-i" name="e-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
  <span id="gux-form-field-help-i" slot="help">
    This is a help message
  </span>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render indicator marks should render component as expected (1) 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render indicator marks should render component as expected (2) 1`] = `
<gux-form-field-select indicator-mark="optional">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="optional"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render indicator marks should render component as expected (3) 1`] = `
<gux-form-field-select indicator-mark="required">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render input attributes should render component as expected (1) 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render input attributes should render component as expected (2) 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select disabled="" id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render input attributes should render component as expected (3) 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" required="" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render label-info should render component as expected 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select aria-describedby="gux-label-info-beta-i" id="gux-form-field-input-i" name="e-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
  <gux-label-info-beta id="gux-label-info-beta-i" slot="label-info">
    <span slot="content">
      This is some tooltip text
    </span>
  </gux-label-info-beta>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render label-position should render component as expected (1) 1`] = `
<gux-form-field-select>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render label-position should render component as expected (2) 1`] = `
<gux-form-field-select label-position="above">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render label-position should render component as expected (3) 1`] = `
<gux-form-field-select label-position="beside">
  <mock:shadow-root>
    <div class="gux-beside gux-form-field-container">
      <div class="gux-beside gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;

exports[`gux-form-field-select #render label-position should render component as expected (4) 1`] = `
<gux-form-field-select label-position="screenreader">
  <mock:shadow-root>
    <div class="gux-form-field-container gux-screenreader">
      <div class="gux-form-field-label gux-screenreader">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
            <gux-icon decorative="" icon-name="custom/chevron-down-small-regular"></gux-icon>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <select id="gux-form-field-input-i" name="lp-1" slot="input">
    <option value="option1">
      Option 1
    </option>
    <option value="option2">
      Option 2
    </option>
    <option value="option3">
      Option 3
    </option>
  </select>
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
</gux-form-field-select>
`;
